### PR TITLE
Various smaller code improvements

### PR DESCRIPTION
**Pull request by iljakuklic**
Therese are smaller improvements, factoring out commonly used functionality etc. See commit messages for more info.

I appreciate this touches many lines of code and will likely conflict with other work in progress stuff. I am happy to discuss when (if at all) is this best merged. Cheers!

### CODE REVIEW

**Review by sinitcin — APPROVED** [review]

File: pallets/utxo/src/lib.rs [hunk 1]
```diff
@@ -179,6 +189,15 @@ pub mod pallet {
                 destination: Destination::CreatePP(code, data),
             }
         }
+
+        /// Create a new output to create a smart contract.
+        pub fn new_call_pp(value: Value, dest_account: AccountId, input: Vec<u8>) -> Self {
```
Comment (altonen) [inline pallets/utxo/src/lib.rs:194]: This is a misleading comment as OP_CALL does not create a new smart contract

Comment (iljakuklic) [reply pallets/utxo/src/lib.rs:194]: Good catch, this is a copy/paste error, will fix

File: pallets/utxo/src/lib.rs [hunk 2]
```diff
@@ -113,13 +113,23 @@ pub mod pallet {
...
+        pub fn new_with_signature(outpoint: H256, sig_script: H512) -> Self {
             Self {
                 outpoint,
                 lock: Vec::new(),
                 witness: (&sig_script[..]).to_vec(),
             }
         }
+
+        /// New input with empty `lock` and `witness` to be filled later.
+        pub fn new_clean(outpoint: H256) -> Self {
```
Comment (muursh) [inline pallets/utxo/src/lib.rs:126]: Is clean the best terminology here? Is new_empty better? Maybe it's just me

**Review by muursh — APPROVED** [review]

File: pallets/utxo/src/lib.rs (hunk 2 above)
Comment (iljakuklic) [reply pallets/utxo/src/lib.rs:126]: Yea I was split between `empty` and `clean` and one other option that I don't remember. Naming things is a nightmare :). Will change that to `empty`.

**Pull request closed without merging**